drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/factPattern/PopupCreator.java

```diff
@@ -106,71 +110,77 @@ public boolean isBindable() {
     /**
      * @param bindable the bindable to set
      */
-    public void setBindable( boolean bindable ) {
+    public void setBindable(boolean bindable) {
         this.bindable = bindable;
     }
 
     /**
      * Display a little editor for field bindings.
      */
-    public void showBindFieldPopup( final FactPattern fp,
-                                    final SingleFieldConstraint con,
-                                    final ModelField[] fields,
-                                    final PopupCreator popupCreator ) {
-        final FormStylePopup popup = new FormStylePopup( GuidedRuleEditorResources.CONSTANTS.AddAField() );
+    public void showBindFieldPopup(final FactPattern fp,
+                                   final SingleFieldConstraint con,
+                                   final ModelField[] fields,
+                                   final PopupCreator popupCreator) {
+        final FormStylePopup popup = new FormStylePopup(GuidedRuleEditorResources.CONSTANTS.AddAField());
 //        popup.setWidth( 500 + "px" );
 //        popup.setHeight( 100 + "px" );
         final HorizontalPanel vn = new HorizontalPanel();
         final TextBox varName = new BindingTextBox();
-        if ( con.getFieldBinding() != null ) {
-            varName.setText( con.getFieldBinding() );
+        if (con.getFieldBinding() != null) {
+            varName.setText(con.getFieldBinding());
         }
-        final Button ok = new Button( HumanReadableConstants.INSTANCE.Set() );
-        vn.add( varName );
-        vn.add( ok );
+        final Button ok = new Button(HumanReadableConstants.INSTANCE.Set());
+        vn.add(varName);
+        vn.add(ok);
 
-        ok.addClickHandler( new ClickHandler() {
-            public void onClick( ClickEvent event ) {
+        ok.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent event) {
                 String var = varName.getText();
-                if ( modeller.isVariableNameUsed( var ) ) {
-                    Window.alert( GuidedRuleEditorResources.CONSTANTS.TheVariableName0IsAlreadyTaken( var ) );
+                if (modeller.isVariableNameUsed(var)) {
+                    Window.alert(GuidedRuleEditorResources.CONSTANTS.TheVariableName0IsAlreadyTaken(var));
+                    return;
+                }
+                String validationTextToDisplay = PatternBindingValidatorUtils.getValidationTextToDisplay(getModeller().getPatternBindingValidators(),
```

Comment:
I'd like to have the check for used variables in one place.. so I'd like to see use of the new validators moved to ```modeller.isVariableNameUsed(var)``` as on L#139 (and necessary changes to return the error message). Probably more refactoring than tagging it on here. Ideally it'd be best to move the whole "duplicate variable name" implementation to use the new ```PatternBindingValidator``` behaviour with core implementations.